docs/topics/gradle/gradle-compiler-options.md

```diff
@@ -15,8 +15,52 @@ in the [Working with command-line compiler](command-line.md) tutorial.
 
 Kotlin compilers have a number of options for tailoring the compiling process.
 
-Using a build script, you can specify additional compilation options. Use the `compilerOptions` property of a Kotlin compilation task for it. 
-For example:
+When you use the Kotlin Gradle plugin in your project, the Gradle DSL allows comprehensive configuration of compiler options. 
+The Gradle DSL is available for [Kotlin Multiplatform](multiplatform-dsl-reference.md) and [JVM/Android](#target-the-jvm) projects.
+
+With the Gradle DSL, you can configure compiler options within the build script at three levels: 
+* **Extension level in top-level blocks:** for all the targets and shared source sets.
+* **Target level in target-level blocks:** for a specific target.
+* **Compilation unit level:** usually a specific compilation task.
+
+![Kotlin compiler options levels](compiler-options-levels.svg){width=700}
+
+The settings at a higher level are used as a convention (default) for a lower level:
+
+* The values of extension compiler options are the default for target compiler options, including shared source sets
+  like `commonMain`, `nativeMain`, and `commonTest`.
+* The values of target compiler options are used as the default for compilation unit (task) compiler options, for
+  example, `compileKotlinJvm` and `compileTestKotlinJvm` tasks.
+
+In turn, configurations made at a lower level override related settings at a higher level:
+
+* Task-level compiler options override related configurations at the target or the extension level.
+* Target-level compiler options override related configurations at the extension level.
+
+To confirm which level of compiler arguments is applied to the compilation, use the `DEBUG` level of Gradle [logging](https://docs.gradle.org/current/userguide/logging.html).
```

Comment:
Maybe add the following:
- for JVM and JS/WASM tasks search log for "Kotlin compiler args:" string
- for Native tasks search log for "Arguments =" string